src/stream.js

```diff
@@ -3,5 +3,97 @@
 const logger = require('./logger');
 
 module.exports.handler = async (event) => {
+	event = decodeEvent({event});
```

Comment:
The core logic here is you want the latest record for each distinct primary key (channel + date). Correct? Using some helper libraries I think you might be able to express this much more compactly:

```javascript
const unmarshalledRecords = event.Records.map(record => AWS.DynamoDB.Converter.unmarshall(record.dynamodb.NewImage));
const groupedRecords= _.groupBy(unmarshalledRecords , record => `${record.channel}.${record.date}`);
for (const [key, value] of Object.entries(groupedRecords)) {
  const latestRecord = _.last(value);
  // do something with latestRecord
}
```

Reply:
I've found that over time my Javascript style has evolved from a more OOP style to a more procedural style. I still use classes occasionally but they are not my default goto solution as they were a few years ago. I think this was because I came from an OOP background originally and applied this lens to my initial Javascript development. I now tend to view my modules more like I would classes. I care more about the interfaces between modules and what data is hidden/exposed in a module. I also value code compactness more than I used to. I prefer anonymous functions/objects more. I find that OOP in Javascript requires extra boilerplate.

My intent is not to criticize what you wrote but I realize swooping in an dropping a couple of lines of code might come across as a passive aggressive critique on what you wrote. If code works and is understandable/maintainable, then it's good code. I also recognize that style is personal and one way is not necessarily better than another. I just wanted to explain my thinking instead of just dropping some code and coming across as smug.

Reply:
Oh yeah, no worries on that! I was hoping you would swoop in and show me a better way or different way. I am coming from the OOP background too and working on the front-end - my coding style some what has suffered. So I really appreciate your time and was looking forward to working with you on this so I could learn.

Reply:
Also, I haven't done much nodeJS development. Almost none.